Makefile

```diff
@@ -1036,3 +1039,88 @@ ceph_cleanup: ## deletes the ceph pod
+	$(eval $(call vars,$@,netattach))
+	bash scripts/gen-netatt.sh
+	oc apply -f ${DEPLOY_DIR}/
+
+##@ METALLB
+.PHONY: metallb
+metallb: export NAMESPACE=metallb-system
+metallb: export INTERFACE=${NNCP_INTERFACE}
+metallb: ## installs metallb operator in the metallb-system namespace
+	$(eval $(call vars,$@,metallb))
+	bash scripts/gen-namespace.sh
+	oc apply -f ${OUT}/${NAMESPACE}/namespace.yaml
+	sleep 2
+	bash scripts/gen-olm-metallb.sh
+	oc apply -f ${OPERATOR_DIR}
+	while ! (oc get pod --no-headers=true -l control-plane=controller-manager -n ${NAMESPACE}| grep "metallb-operator-controller"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} --for condition=Ready -l control-plane=controller-manager --timeout=180s
+	while ! (oc get pod --no-headers=true -l component=webhook-server -n ${NAMESPACE}| grep "metallb-operator-webhook"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} --for condition=Ready -l component=webhook-server --timeout=180s
+	oc apply -f ${DEPLOY_DIR}/deploy_operator.yaml
+	while ! (oc get pod --no-headers=true -l component=speaker -n ${NAMESPACE} | grep "speaker"); do sleep 10; done
+	oc wait pod -n ${NAMESPACE} -l component=speaker --for condition=Ready --timeout=180s
+
+.PHONY: metallb_config
+metallb_config: export NAMESPACE=metallb-system
+metallb_config: export INTERFACE=${NNCP_INTERFACE}
+metallb_config: ## creates the IPAddressPools and l2advertisement resources
+	$(eval $(call vars,$@,metallb))
+	bash scripts/gen-olm-metallb.sh
+	oc apply -f ${DEPLOY_DIR}/ipaddresspools.yaml
+	oc apply -f ${DEPLOY_DIR}/l2advertisement.yaml
+
+##@ EDPM
+.PHONY: edpm_deploy
+edpm_deploy: export OVN_METADATA_AGENT_TRANSPORT_URL_USER=$(shell oc get secret rabbitmq-default-user -o json | jq -r .data.username | base64 -d)
+edpm_deploy: export OVN_METADATA_AGENT_TRANSPORT_URL_PASSWORD=$(shell oc get secret rabbitmq-default-user -o json | jq -r .data.password | base64 -d)
+edpm_deploy: export OVN_METADATA_AGENT_SB_CONNECTION=$(shell oc get ovndbcluster ovndbcluster-sb -o json | jq -r .status.dbAddress)
+edpm_deploy: export EDPM_OVN_DBS=$(shell oc get ovndbcluster ovndbcluster-sb -o json | jq -r '.status.networkAttachments."openstack/internalapi"[0]')
+edpm_deploy: export OVN_METADATA_AGENT_NOVA_METADATA_HOST=127.0.0.1
+edpm_deploy: export OVN_METADATA_AGENT_PROXY_SHARED_SECRET=12345678
+edpm_deploy: export OVN_METADATA_AGENT_BIND_HOST=127.0.0.1
+edpm_deploy: export CHRONY_NTP_SERVER=clock.redhat.com
+edpm_deploy: ## Creates OpenStackDataPlaneNode resource to run deployment job
+	$(eval $(call vars,$@,edpm_deploy))
+	bash scripts/gen-edpm.sh
+	oc apply -f ${DEPLOY_DIR}/
+	# TODO
+	# oc wait openstackdataplanenode -n ${NAMESPACE} -l component=openstackdataplanenode --for condition=Ready --timeout=600s
+
+.PHONY: edpm_deploy_cleanup
+edpm_deploy_cleanup: ## Deletes the OpenStackDataPlaneNode resource, does not delete the edpm node
+	$(eval $(call vars,$@,edpm_deploy))
+	oc delete --ignore-not-found=true -f ${DEPLOY_DIR}/
```

Comment:
not completely sure if edpm deployment under devsetup dir as it is now with https://github.com/openstack-k8s-operators/install_yamls/blob/master/devsetup/Makefile#L60

Reply:
I might remove this and test integrate with current `make edpm_play` from devsetup